canonical-web-and-design/maas-ui
# feat(ui): add missing keys and actions to subnet slice

## Done

- This branch adds eventErrors and statuses to the subnet slice, and includes reducers for the 'scan' action

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to `/MAAS/r/networks` and click "Add > Subnet"
- Enter some bogus IP data and submit the form
- You should get an error. Check in Redux DevTools that the error shows up under `subnet.eventErrors` as well as `subnet.errors`
- Go to the network tab of a machine (or any other React page where subnets are fetched)
- Check that `subnet.statuses` is populated correctly

## Fixes

Fixes canonical-web-and-design/app-tribe#640
